xcp_d/utils/concantenation.py

```diff
@@ -180,149 +191,210 @@ def concatenate_nifti(subid, fmridir, outputdir, ses=None, work_dir=None):
-    # filex to be concatenated
-
+    # files to be concatenated
     datafile = [
-        '_atlas-Glasser_desc-timeseries_bold.tsv',
-        '_atlas-Gordon_desc-timeseries_bold.tsv',
-        '_atlas-Schaefer117_desc-timeseries_bold.tsv',
-        '_atlas-Schaefer617_desc-timeseries_bold.tsv',
-        '_atlas-Schaefer217_desc-timeseries_bold.tsv',
-        '_atlas-Schaefer717_desc-timeseries_bold.tsv',
-        '_atlas-Schaefer317_desc-timeseries_bold.tsv',
-        '_atlas-Schaefer817_desc-timeseries_bold.tsv',
-        '_atlas-Schaefer417_desc-timeseries_bold.tsv',
-        '_atlas-Schaefer917_desc-timeseries_bold.tsv',
-        '_atlas-Schaefer517_desc-timeseries_bold.tsv',
-        '_atlas-Schaefer1017_desc-timeseries_bold.tsv',
-        '_atlas-subcortical_desc-timeseries_bold.tsv',
-        '_desc-framewisedisplacement_bold.tsv', '_desc-residual_bold.nii.gz',
-        '_desc-residual_smooth_bold.nii.gz'
+        '_desc-filtered_motion.tsv',  # Must come first to set motion_suffix
+        '_desc-denoised_bold.nii.gz',
+        '_desc-denoisedSmoothed_bold.nii.gz',
+        '_atlas-Glasser_timeseries.tsv',
+        '_atlas-Gordon_timeseries.tsv',
+        '_atlas-Schaefer117_timeseries.tsv',
+        '_atlas-Schaefer617_timeseries.tsv',
+        '_atlas-Schaefer217_timeseries.tsv',
+        '_atlas-Schaefer717_timeseries.tsv',
+        '_atlas-Schaefer317_timeseries.tsv',
+        '_atlas-Schaefer817_timeseries.tsv',
+        '_atlas-Schaefer417_timeseries.tsv',
+        '_atlas-Schaefer917_timeseries.tsv',
+        '_atlas-Schaefer517_timeseries.tsv',
+        '_atlas-Schaefer1017_timeseries.tsv',
+        '_atlas-subcortical_timeseries.tsv',
     ]
 
     if ses is None:
-        all_func_files = glob.glob(str(outputdir) + '/' + subid + '/func/*')
-        fmri_files = str(fmridir) + '/' + subid + '/func/'
-        figure_files = str(outputdir) + '/' + subid + '/figures/'
+        all_func_files = glob.glob(os.path.join(outputdir, subid, 'func', '*'))
+        func_dir = os.path.join(fmridir, subid, 'func')
+        figures_dir = os.path.join(outputdir, subid, 'figures')
     else:
-        all_func_files = glob.glob(
-            str(outputdir) + '/' + subid + '/ses-' + str(ses) + '/func/*')
-        fmri_files = str(fmridir) + '/' + subid + '/ses-' + str(ses) + '/func/'
-        figure_files = str(outputdir) + '/' + subid + '/figures/'
-
-    fmri_files = str(fmri_files)
+        all_func_files = glob.glob(os.path.join(outputdir, subid, f'ses-{ses}', 'func', '*'))
+        func_dir = os.path.join(fmridir, subid, f'ses-{ses}', 'func')
+        figures_dir = os.path.join(outputdir, subid , 'figures')
 
     # extract the task list
+    denoised_bold_files = fnmatch.filter(all_func_files, '*_desc-denoised_bold.nii.gz')
     tasklist = [
-        os.path.basename(j).split('task-')[1].split('_')[0]
-        for j in fnmatch.filter(all_func_files, '*_desc-residual_bold.nii.gz')
+        denoised_bold_file.split('task-')[-1].split('_')[0]
```

Comment:
Is there a reason for changing this? I tried locally but either seems to work for me

Reply:
The main reason I changed it was because `j` isn't an informative variable name. The `os.path.basename` is unnecessary because we're grabbing the split from the last instance of `task-` in the string, so it's robust to previous cases of `task-` in the full path.

Reply:
got it